File: code_snippets/go/develop/myservice.go
package develop

import (
	"context"
	"fmt"

	restate "github.com/restatedev/sdk-go"
	server "github.com/restatedev/sdk-go/server"
)

type MyVirtualObject struct{}

func (MyVirtualObject) MyHandler(ctx restate.ObjectContext, greeting string) (string, error) {
	return fmt.Sprintf("%s %s!", greeting, ctx.Key()), nil
}

func (MyVirtualObject) MyConcurrentHandler(ctx restate.ObjectSharedContext, greeting string) (string, error) {
	return fmt.Sprintf("%s %s!", greeting, ctx.Key()), nil
}

func serveMyVirtualObject() error {
	return server.NewRestate().
		Bind(restate.Reflect(MyVirtualObject{})).
		Start(context.Background(), "0.0.0.0:9080")
}

File: code_snippets/ts/src/develop/my_service.ts
import * as restate from "@restatedev/restate-sdk";
import {Context} from "@restatedev/restate-sdk";

const myService = restate.service({
    name: "MyService",
    handlers: {
        myHandler: async (ctx: Context, greeting: string) => {
            return `${greeting}!`;
        },
    }
})

// <start_api_export>
export const MyService: typeof myService = { name: "MyService" };
// <end_api_export>

restate
    .endpoint()
    .bind(myService)
    .listen();